stages/org.osbuild.coreos.live-iso

```diff
@@ -296,12 +303,23 @@ def main(inputs, devices, tree, options):
     # If mounted via the manifest, the stage begins with mounts already in place,
     # but osmet also performs a mount operation, leading to conflicts due to duplicate
     # filesystem UUIDs. Perform the manual mount only after the osmet stage
-    subprocess.check_call(['mount', '-o', 'rw', img_metal + 'p4', tmp_squashfs_dir])
-    subprocess.check_call(['mount', '-o', 'rw', img_metal + 'p3',
+    subprocess.check_call(['mount', '-o', 'ro', img_metal + 'p4', tmp_squashfs_dir])
+    subprocess.check_call(['mount', '-o', 'ro', img_metal + 'p3',
                            os.path.join(tmp_squashfs_dir, 'boot')])
     if basearch in ['x86_64', 'aarch64']:
-        subprocess.check_call(['mount', '-o', 'rw', img_metal + 'p2',
+        subprocess.check_call(['mount', '-o', 'ro', img_metal + 'p2',
                                os.path.join(tmp_squashfs_dir, 'boot/efi')])
+    # Mount the filesystem using OverlayFS in a read-only mode to ensure the image
+    # remains unchanged and consistent
+    # Note OverlayFS does not support traversing across filesystem boundaries
+    # That's the reason we need to mounts here.
+    subprocess.check_call(['mount', '-t', 'overlay', 'overlay', '-o',
+                           f'lowerdir={tmp_squashfs_dir},upperdir={tmp_upperdir_root},workdir={tmp_workdir_root}',
+                           f'{tmp_merged_dir}'])
+
+    subprocess.check_call(['mount', '-t', 'overlay', 'overlay', '-o',
+                           f'lowerdir={tmp_squashfs_dir}/boot,upperdir={tmp_upperdir_boot},workdir={tmp_workdir_boot}',
+                           f'{tmp_merged_dir}/boot'])
     umount_needed = True
```

Comment:
I think in theory this should now be two variables. E.g. `squashfs_dir_umount_needed` and `merged_dir_umount_needed`. So that if we error out between the two, we're not going to try to `umount` something that wasn't mounted at all (and cause even more errors).